### PR TITLE
Handle indexed slice empty shapes in IndexedSlices gradients correctly.

### PR DESCRIPTION
**Pull request by arvindsun**
Potential fix for https://github.com/tensorflow/tensorflow/issues/31962

I ran into this after updating to the 2.0 final release. I am training a LSTM network using .fit() passing in Numpy arrays as input.

### CODE REVIEW

**Comment by arvindsun** [comment]
@googlebot I signed it!

**Comment by arvindsun** [comment]
This is really small minor fix which fixes TF2.0 usage with numpy datasets. I am happy to get into this more, but it would be really good if the next patch release had this fix.

**Review by alextp — CHANGES_REQUESTED** [review]
Can you add a unit test which fails before this change and passes after?

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by arvindsun** [comment]
I will get the unittests done this weekend.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by mjlbach** [comment]
This fixes #31962 for me. Thanks! Hope this gets upstreamed.

**Review by mdanatg — APPROVED** [review]
It would be great to add a test, but it does seem to unblock at least one user.
